neasqc_wp61/models/quantum/dim_reduction.py

```diff
@@ -0,0 +1,270 @@
+
+"""
+DimReduction
+============
+Module containing the base class for performing dimensionality reduction
+
+"""
+from abc import ABC, abstractmethod
+import pandas as pd
+import sklearn.decomposition as skd
+import sklearn.manifold as skm
+import numpy as np 
+import umap
+
+class DimReduction(ABC):
+    """
+    Base class for dimensionality reduction of 
+    vectors representing sentences. 
+    """
+    def __init__(
+        self, dataset : pd.DataFrame, dim_out : int
+    )-> None:
+        """
+        Initialises the dimensionality reduction class
+        
+        Parameters
+        ----------
+        dataset : pd.DataFrame
+            Pandas dataset. Each row of the dataset corresponds
+            to a sentence. The dataset must contain one column named
+            'sentence_vector', with the vector representation 
+            of each sentence.
+        dim_out : int 
+            Desired output dimension of the vectors
+        """
+
+        self.dataset = dataset
+        if 'sentence_embedding' not in self.dataset:
+            raise ValueError('sentence vector not present in the dataset')
+        self.sentence_vectors = dataset['sentence_embedding'].to_list()
+        self.dim_out = dim_out
+
+    @abstractmethod
+    def fit(self) -> None:
+        """
+        Fits the dataset to output vectors with the desired dimension
```

Comment:
Please change the name!
Functions called 'fit' have a very specific purpose and its not that one. It's very confusing for someone to coma accross a function called fit that isn't about optimisation.